packages/mdc-switch/_mixins.scss

```diff
@@ -71,8 +154,146 @@
   }
 }
 
-@mixin mdc-switch-toggled-off-ripple-color($color) {
+@mixin mdc-switch-toggled-off-ripple-color($color, $query: mdc-feature-all()) {
   &:not(.mdc-switch--checked) .mdc-switch__thumb-underlay {
-    @include mdc-states($color);
+    @include mdc-states($color, false, $query);
+  }
+}
+
+//
+// Private
+//
+
+// Structure
+@mixin mdc-switch-base_ {
+  display: inline-block;
+  position: relative;
+  outline: none;
+  user-select: none;
+}
+
+@mixin mdc-switch__track_($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-feature-targets($feat-structure) {
+    box-sizing: border-box;
+    width: $mdc-switch-track-width;
+    height: $mdc-switch-track-height;
+    border: 1px solid;
+    border-radius: $mdc-switch-track-height / 2;
+    opacity: .38;
+  }
+
+  @include mdc-feature-targets($feat-animation) {
+    transition:
+      mdc-switch-transition(opacity),
+      mdc-switch-transition(background-color),
+      mdc-switch-transition(border-color);
+  }
+}
+
+@mixin mdc-switch__thumb-underlay_($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-states($mdc-switch-baseline-theme-color, false, $query);
```

Comment:
Move `mdc-states` after `mdc-ripple-surface` (`mdc-ripple-surface` first is a common pattern across the packages that use it)